src/components/AnonymousChat/MessageChannelItem.tsx

```diff
@@ -1,26 +1,68 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// eslint-disable-next-line no-use-before-define
 import * as React from 'react';
 
 import BaseChannelItem from './BaseChannelItem';
+import useChatUtilsHook from '../../hooks/core/chat/useChatUtilsHook';
+import useProfileHook from '../../hooks/core/profile/useProfileHook';
 import {BaseChannelItemTypeProps} from '../../../types/component/AnonymousChat/BaseChannelItem.types';
+import {ChannelList} from '../../../types/database/schema/ChannelList.types';
+import {Context} from '../../context';
 import {MessageChannelItemProps} from '../../../types/component/AnonymousChat/MessageChannelItem.types';
 import {calculateTime} from '../../utils/time';
+import {getChatName} from '../../utils/string/StringUtils';
 
 const MessageChannelItem: (props: MessageChannelItemProps) => React.ReactElement = ({
   item,
-  onChannelPressed
+  onChannelPressed,
+  hasFollowButton = false,
+  handleFollow
 }) => {
+  const {signedProfileId} = useProfileHook();
+
+  const determineChatType = (data: ChannelList) => {
+    const isAnonymous = Boolean(data?.channelType === 'ANON_PM');
+    if (!isAnonymous) return BaseChannelItemTypeProps.SIGNED_PM;
+    return BaseChannelItemTypeProps.ANON_PM;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `determineChatType` returns a string based on the condition `data?.channelType === 'ANON_PM'`. This could be simplified using a ternary operator.
```diff
-  const determineChatType = (data: ChannelList) => {
-    const isAnonymous = Boolean(data?.channelType === 'ANON_PM');
-    if (!isAnonymous) return BaseChannelItemTypeProps.SIGNED_PM;
-    return BaseChannelItemTypeProps.ANON_PM;
-  };
+  const determineChatType = (data: ChannelList) =>
+    data?.channelType === 'ANON_PM' ? BaseChannelItemTypeProps.ANON_PM : BaseChannelItemTypeProps.SIGNED_PM;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->